include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -320,6 +321,32 @@ Vinecop::select_all(Eigen::MatrixXd data,
   finalize_fit(selector);
 }
 
+//! @brief automatically fits and selects a vine copula model.
+//!
+//! Selection of the structure is performed using the algorithm of
+//! Dissmann, J. F., E. C. Brechmann, C. Czado, and D. Kurowicka (2013).
+//! *Selecting and estimating regular vine copulae and application to
+//! financial returns.* Computational Statistics & Data Analysis, 59 (1),
+//! 52-69.
+//!
+//! @details When at least one variable is discrete, two types of "observations"
+//! are required: the first \f$ n \times d \f$ block contains realizations of
+//! \f$ F_Y(Y), F_X(X) \f$; the second \f$ n \times d \f$ block contains
+//! realizations of \f$ F_Y(Y^-), F_X(X^-), ... \f$. The minus indicates a
+//! left-sided limit of the cdf. For continuous variables the left limit and the
+//! cdf itself coincide. For, e.g., an integer-valued variable, it holds \f$
+//! F_Y(Y^-) = F_Y(Y - 1) \f$. Continuous variables in the second block can
+//! be omitted.
+//!
+//! @param data \f$ n \times (d + k) \f$ or \f$ n \times 2d \f$ matrix of
+//!   observations, where \f$ k \f$ is the number of discrete variables.
+//! @param controls the controls to the algorithm (see FitControlsVinecop).
+inline void
+Vinecop::select_all(Eigen::MatrixXd data, const FitControlsVinecop& controls)
+{
+  select(data, controls);
+}
+
```

Comment:
Shouldn't we set `vine_struct_ = RVineStructure(tools_stl::seq_int(1, d_), static_cast<size_t>(0))` first?